Tests for misc tables fail with pandas v3.0.2

`pandas` v3.0.2 was released and now `tests/test_misc_table.py` fails with:

```
____________________________________________________________________________ test_load_old_pickle ____________________________________________________________________________

tmpdir = local('/tmp/pytest-of-hwierstorf/pytest-74/test_load_old_pickle0')

    def test_load_old_pickle(tmpdir):
        # We have stored string dtype as object dtype before
        # and have to fix this when loading old PKL files from cache.
        # This does only affect columns
        # as there was no MiscTable available.
    
        # Create PKL file containing strings as object
        y = pd.Series(["c"], dtype="object", name="column")
        index = pd.Index(["i"], dtype="object", name="idx")
    
        db = audformat.testing.create_db(minimal=True)
        db["misc"] = audformat.MiscTable(index)
        db.schemes["column"] = audformat.Scheme(audformat.define.DataType.OBJECT)
        db["misc"]["column"] = audformat.Column(scheme_id="column")
        db["misc"]["column"].set(y.values)
        db_root = tmpdir.join("db")
        db.save(db_root, storage_format="pkl")
    
        # Change scheme dtype to string and store header again
        db.schemes["column"] = audformat.Scheme(audformat.define.DataType.STRING)
        db.save(db_root, header_only=True)
    
        # Load and check that dtype is string
        db_new = audformat.Database.load(db_root)
        assert db_new.schemes["column"].dtype == audformat.define.DataType.STRING
>       assert db_new["misc"].df["column"].dtype == "string"
E       AssertionError: assert dtype('O') == 'string'
E        +  where dtype('O') = idx\ni    c\nName: column, dtype: object.dtype

/home/hwierstorf/git/audeering/audformat/tests/test_misc_table.py:1361: AssertionError
```

See also https://github.com/audeering/audformat/actions/runs/24197025175/job/70629819798?pr=526

--- a/tests/test_misc_table.py
+++ b/tests/test_misc_table.py
@@ -1,3 +1,5 @@
+import os
+
 import numpy as np
 import pandas as pd
 import pytest
@@ -1348,12 +1350,15 @@ def test_load_old_pickle(tmpdir):
     db.schemes["column"] = audformat.Scheme(audformat.define.DataType.OBJECT)
     db["misc"]["column"] = audformat.Column(scheme_id="column")
     db["misc"]["column"].set(y.values)
-    db_root = tmpdir.join("db")
-    db.save(db_root, storage_format="pkl")
+    db_root = str(tmpdir.join("db"))
 
     # Change scheme dtype to string and store header again
+    # and write bare DataFrame pickle
+    # (simulating pre-1.4.0 format without version wrapper)
     db.schemes["column"] = audformat.Scheme(audformat.define.DataType.STRING)
     db.save(db_root, header_only=True)
+    df = pd.DataFrame({"column": y}, index=index)
+    df.to_pickle(os.path.join(db_root, "db.misc.pkl"), protocol=4)
 
     # Load and check that dtype is string
     db_new = audformat.Database.load(db_root)

